packages/code-snippet/src/CodeSnippetWidget.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2018-2020 IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import '../style/index.css';
+import React from 'react';
+
+import { IDocumentManager } from '@jupyterlab/docmanager';
+import { IRenderMimeRegistry } from '@jupyterlab/rendermime';
+import { ReactWidget, UseSignal } from '@jupyterlab/apputils';
+import { Message } from '@phosphor/messaging';
+import { Signal } from '@phosphor/signaling';
+
+import { CodeSnippetManager, ICodeSnippet } from './CodeSnippet';
+import { ExpandableComponent } from './ExpandableComponent';
+
+/**
+ * The CSS class added to code snippet widget.
+ */
+const CODE_SNIPPETS_CLASS = 'elyra-CodeSnippets';
+const CODE_SNIPPETS_HEADER_CLASS = 'elyra-codeSnippetsHeader';
+const CODE_SNIPPET_ITEM = 'elyra-codeSnippet-item';
+const BUTTON_CLASS = 'elyra-button';
+const COPY_ICON_CLASS = 'elyra-copy-icon';
+const INSERT_ICON_CLASS = 'elyra-add-icon';
+const CODE_SNIPPET_NAME_CLASS = 'elyra-codeSnippet-name';
+const CODE_SNIPPET_BUTTONS_WRAPPER_CLASS = 'elyra-codeSnippet-buttons';
+
+/**
+ * A widget for code-snippet.
+ */
+interface ICodeSnippetProps {
+  codeSnippets: ICodeSnippet[];
+}
+class CodeSnippetTable extends React.Component<ICodeSnippetProps> {
+  // TODO: Use code mirror to display code
+  // TODO: implement copy to clipboard command
+  // TODO: implement insert code to file editor command (first check for code language matches file editor kernel language)
+
+  render(): React.ReactElement {
+    const renderCodeSnippet = (codeSnippet: ICodeSnippet): JSX.Element => {
```

Comment:
This can be declared as private outside the render function for better readability